### ISSUE TITLE
MA0131 is reported for generic parameters

### ISSUE THREAD

**Issue by rsking** [meziantou/Meziantou.Analyzer#553]
### Version of the Meziantou.Analyzer NuGet package

2.0.62

### Rule Identifier

MA0131

### Target Framework

net6.0, net7.0

### C# Language version

C#11

### Description

Generics are causing MA0131 to throw. We do not know if this is a non-nullable value, which is why the check is being done.

### Reproduction Steps

Minimal code:

```c#
static void CheckMA0131Incorrect<T>(T value)
{
    ArgumentNullException.ThrowIfNull(value);
}
```

### Other information

Adding a `class` constraint on the generic does make it go away, but I use this in a way that constrains the generic to an interface. I do not know at compile time if a struct or object will be passed in.

Even if I put a nullable annotation on the generic (i.e. `T?`), it still raises the issue.

**Comment by meziantou** [comment]
Thanks for reporting the issue! The fix is available in [2.0.63](https://www.nuget.org/packages/Meziantou.Analyzer/2.0.63)
